### PR TITLE
Improvement of manifest generation

### PR DESCRIPTION
**Pull request by cvgaviao**
fixed wrongly changed configuration of core and compile manifest.
improved the manifest generation integrating maven-bundle-plugin with m2e

### CODE REVIEW

File: drools-compiler/pom.xml [hunk 1]
```diff
@@ -10,7 +10,7 @@
   </parent>
 
   <artifactId>drools-compiler</artifactId>
-
+  <packaging>bundle</packaging>
```
Comment (ge0ffrey) [inline drools-compiler/pom.xml:13]: Interesting idea.
Could you verify that from a consumer's point of view, drools-compiler is still just a jar and nothing has changed?

**Comment by cvgaviao** [comment]
related to https://github.com/droolsjbpm/droolsjbpm-build-bootstrap/pull/21

**Comment by ge0ffrey** [comment]
Great stuff.

jbpm and drools should handle the OSGi stuff in a similar matter. Would you be willing to provide a similar pull request for jbpm's master? https://github.com/droolsjbpm/jbpm
Also these might have some osgi config:
  https://github.com/droolsjbpm/droolsjbpm-knowledge
  https://github.com/droolsjbpm/droolsjbpm-build-distribution

**Comment by cvgaviao** [comment]
yep... is on the way ;)

**Comment by ge0ffrey** [comment]
Great :) Beware of droolsjbpm-build-distribution: it's at the end of the build chain and jenkins isn't looking stable due to the uberfire prototyping. https://hudson.jboss.org/hudson/view/Drools%20jBPM/
Everything in drools and jbpm must be verified to work, but for droolsjbpm-build-distribution I'll be far more lenient currently... (better to merge any changes needed there now, if it doesn't build/run worse than originally).

**Comment by cvgaviao** [comment]
well, I don't use the bundles from droolsjbpm-build-distribution. 
As I use Equinox, I created my own p2 repository with Tycho that contains all the bundles needed by drools and jbpm.

Btw, protobuf 2.5 will be released soon, it was already osgified...

**Comment by cvgaviao** [comment]
I'm looking now at new kie projects at https://github.com/droolsjbpm/droolsjbpm-knowledge.
The complicated thing (osgi point of view) is that there are equal packages at kie-api and kie-internal, that is not good...

**Comment by ge0ffrey** [comment]
" is that there are equal packages at kie-api and kie-internal, that is not good" - for 6 there's work being doing to namespace all artifacts by a unique namespace. (for example drools-compiler's classes start with org.drools.compiler).

Good hint about protobuf, I poked Edson/Mark to see if we can upgrade and remove our hack in build-distribution.
build-distribution just contains the OSGi testcase really. But it's dreadfully brittle and unstable that testcase. If you know a better way to smoke test that drools starts up in an OSGi container, it's welcome :)

**Comment by cvgaviao** [comment]
well, I used to run junit tests inside our tycho equinox product build. but that will not suite well for drools/jbpm team.

I could invest some time and to create a integration test project using pax-exam, this way we can test on both equinox and felix containers. wdyt ?

File: drools-compiler/pom.xml (hunk 1 above)
Comment (cvgaviao) [inline drools-compiler/pom.xml:13]: yep, its just a normal jar artifact. that will just trigger the manisfest generation at eclipse pde and will be integrated with m2e builder. the manifest will be generated at target folder (as any other artifact) and pde will know that project is a bundle.

**Comment by cvgaviao** [comment]
Geoffrey, one of the benefits of the proposed changes in manifest is that now it much easier to identify problems in manifest generation and anomalies at package structure. 
see this log:
<code>
12/11/12, 1:59:36 PM GMT-03:00: [WARN] Manifest org.drools:drools-persistence-jpa:bundle:6.0.0-SNAPSHOT : Split package org/drools/reteoo/compiled
Use directive -split-package:=(merge-first|merge-last|error|first) on Export/Private Package instruction to get rid of this warning
Package found in   [Jar:drools-core, Jar:drools-compiler]
Reference from     /Users/cvgaviao/Development/Development_Drools&Jbpm/drools/drools-compiler/target/classes
Classpath          [Jar:., Jar:kie-api, Jar:drools-core, Jar:mvel2, Jar:kie-internal, Jar:drools-compiler, Jar:antlr, Jar:stringtemplate, Jar:xstream, Jar:xmlpull, Jar:xpp3_min, Jar:jaxb-impl, Jar:jaxb-xjc, Jar:org.osgi.core, Jar:org.osgi.compendium, Jar:hibernate-jpa-2.0-api, Jar:protobuf-java, Jar:dom4j, Jar:xml-apis, Jar:javassist, Jar:jta, Jar:slf4j-api]
12/11/12, 1:59:36 PM GMT-03:00: [WARN] Manifest org.drools:drools-persistence-jpa:bundle:6.0.0-SNAPSHOT : Split package org/drools/rule/builder/dialect/asm
Use directive -split-package:=(merge-first|merge-last|error|first) on Export/Private Package instruction to get rid of this warning
Package found in   [Jar:drools-core, Jar:drools-compiler]
Reference from     /Users/cvgaviao/Development/Development_Drools&Jbpm/drools/drools-compiler/target/classes
Classpath          [Jar:., Jar:kie-api, Jar:drools-core, Jar:mvel2, Jar:kie-internal, Jar:drools-compiler, Jar:antlr, Jar:stringtemplate, Jar:xstream, Jar:xmlpull, Jar:xpp3_min, Jar:jaxb-impl, Jar:jaxb-xjc, Jar:org.osgi.core, Jar:org.osgi.compendium, Jar:hibernate-jpa-2.0-api, Jar:protobuf-java, Jar:dom4j, Jar:xml-apis, Jar:javassist, Jar:jta, Jar:slf4j-api]
12/11/12, 1:59:36 PM GMT-03:00: [WARN] Manifest org.drools:drools-persistence-jpa:bundle:6.0.0-SNAPSHOT : Did not find matching referal for !org.drools.*
12/11/12, 1:59:42 PM GMT-03:00: [WARN] Manifest org.drools:drools-templates:bundle:6.0.0-SNAPSHOT : Split package org/drools/reteoo/compiled
Use directive -split-package:=(merge-first|merge-last|error|first) on Export/Private Package instruction to get rid of this warning
Package found in   [Jar:drools-core, Jar:drools-compiler]
Reference from     /Users/cvgaviao/Development/Development_Drools&Jbpm/drools/drools-compiler/target/classes
Classpath          [Jar:., Jar:drools-core, Jar:mvel2, Jar:kie-api, Jar:kie-internal, Jar:slf4j-api, Jar:drools-compiler, Jar:antlr-runtime, Jar:antlr, Jar:stringtemplate, Jar:antlr, Jar:ecj, Jar:xstream, Jar:xmlpull, Jar:xpp3_min]
12/11/12, 1:59:42 PM GMT-03:00: [WARN] Manifest org.drools:drools-templates:bundle:6.0.0-SNAPSHOT : Split package org/drools/rule/builder/dialect/asm
Use directive -split-package:=(merge-first|merge-last|error|first) on Export/Private Package instruction to get rid of this warning
Package found in   [Jar:drools-core, Jar:drools-compiler]
Reference from     /Users/cvgaviao/Development/Development_Drools&Jbpm/drools/drools-compiler/target/classes
Classpath          [Jar:., Jar:drools-core, Jar:mvel2, Jar:kie-api, Jar:kie-internal, Jar:slf4j-api, Jar:drools-compiler, Jar:antlr-runtime, Jar:antlr, Jar:stringtemplate, Jar:antlr, Jar:ecj, Jar:xstream, Jar:xmlpull, Jar:xpp3_min]
12/11/12, 1:59:42 PM GMT-03:00: [WARN] Manifest org.drools:drools-templates:bundle:6.0.0-SNAPSHOT : Did not find matching referal for !org.drools.*
12/11/12, 2:09:41 PM GMT-03:00: [WARN] Manifest org.drools:drools-templates:bundle:6.0.0-SNAPSHOT : Split package org/drools/reteoo/compiled
Use directive -split-package:=(merge-first|merge-last|error|first) on Export/Private Package instruction to get rid of this warning
Package found in   [Jar:drools-core, Jar:drools-compiler]
Reference from     /Users/cvgaviao/Development/Development_Drools&Jbpm/drools/drools-compiler/target/classes
Classpath          [Jar:., Jar:drools-core, Jar:mvel2, Jar:kie-api, Jar:kie-internal, Jar:slf4j-api, Jar:drools-compiler, Jar:antlr-runtime, Jar:antlr, Jar:stringtemplate, Jar:antlr, Jar:ecj, Jar:xstream, Jar:xmlpull, Jar:xpp3_min]
12/11/12, 2:09:41 PM GMT-03:00: [WARN] Manifest org.drools:drools-templates:bundle:6.0.0-SNAPSHOT : Split package org/drools/rule/builder/dialect/asm
Use directive -split-package:=(merge-first|merge-last|error|first) on Export/Private Package instruction to get rid of this warning
Package found in   [Jar:drools-core, Jar:drools-compiler]
Reference from     /Users/cvgaviao/Development/Development_Drools&Jbpm/drools/drools-compiler/target/classes
Classpath          [Jar:., Jar:drools-core, Jar:mvel2, Jar:kie-api, Jar:kie-internal, Jar:slf4j-api, Jar:drools-compiler, Jar:antlr-runtime, Jar:antlr, Jar:stringtemplate, Jar:antlr, Jar:ecj, Jar:xstream, Jar:xmlpull, Jar:xpp3_min]
12/11/12, 2:12:09 PM GMT-03:00: [INFO] Using org.eclipse.m2e.jdt.JarLifecycleMapping lifecycle mapping for MavenProject: org.drools:drools-templates:6.0.0-SNAPSHOT @ /Users/cvgaviao/Development/Development_Drools&Jbpm/drools/drools-templates/pom.xml.
</code>

**Comment by ge0ffrey** [comment]
Merged + some changes + I 've also used the same tactic on kie-api and kie-internal (and knowledge-api and knowledge-internal-api). Thanks!

I 'll now merge the pull request on jbpm too.

**Comment by ge0ffrey** [comment]
Christiano,

Some open questions:
- Do we still need those build.properties files? For example: https://github.com/droolsjbpm/drools/blob/master/drools-core/build.properties
- How do we get to see that log? Mark wants to resolve the split packages problem soon.

**Comment by cvgaviao** [comment]
Hi Geoffrey,
1) I don't think so. This file is used when you have a bundle (that we call manifest-first) what is not the case of drools-core(it is pom-first).

2) the log is get in the m2e console inside eclipse. You can use a Project/Clean command that will clear and call the build again... If you and Mark wanted I could handle this as soon he refactored the packages (including the test packages)

**Comment by ge0ffrey** [comment]
Thanks for the answers.
1) I 'll remove them in that case
2) That would probably be welcome. (I won't handle that issue.) Mark's nickname on irc is conan, on the channel drools and drools-dev. Keep an eye out when the packages have been split and ping him then: any help especially with OSGi is welcome :)

**Pull request closed without merging**